lib/puppet/functions/cd4pe_deployments/get_pipeline_trigger_event.rb

```diff
@@ -24,18 +24,16 @@
   end
 
   def get_pipeline_trigger_event(repo_name, pipeline_id, commit_sha)
```

Comment:
You mention deprecating this method, but @kreeuwijk added this in https://github.com/puppetlabs/puppetlabs-cd4pe_deployments/pull/49 for a specific reason so you might want to check in with him first. I see some value in having a method to get the latest one vs. all of them. We could rename the function though.

Reply:
I'm fine with this, I looked at the code changes and this doesn't break anything. I'm happy to switch to `get_pipeline_trigger_events()` and do the `.first` myself in the plan code .